proto/caraml/upi/v1/table.proto

```diff
@@ -0,0 +1,53 @@
+syntax = "proto3";
+
+package caraml.upi.v1;
+
+import "caraml/upi/v1/value.proto";
+
+
+// Table represents a 2D data structure that has one or more columns 
+// with potentially different types
+message Table {
+    // Table's name
+    string name = 1;
+    // Columns stores schema informations of all columns in the table.
+    repeated Column columns = 2;
+    // Rows stores list of row values in the table.
+    repeated Row rows = 3;
+}
+
+// Column represent a column definition within a table
+message Column {
+    // Column's name
+    string name = 1;
+    // Column's type
+    Type type = 2;
+}
+
+// Row represents list of value stored within a row of a table
+message Row {
+    // Row ID is defined by the client and can be used
+    // to join a prediction row with the prediction result,
+    // and to track predictions generated by multiple models.
+    // The user is expected include row ID (along with prediction ID)
+    // when calling the observations API so that predictions
+    // and observations can be joined. 
+    string row_id = 1;
```

Comment:
how should `row_id` be used for transformer input tables? will the platform validate `row_id` is unique within each table? should we somehow differentiate the simpler and more common scenario where the transformer_input table has the same row_id as the prediction_table?

Reply:
> how should row_id be used for transformer input tables?
This become optional in the context of `Table` in `transformer_input`. If it exists, an additional column `row_id` will present in the table when standard transformer imports it. @tiopramayudi will share short RFC on how standard transformer interact with this. 

> should we somehow differentiate the simpler and more common scenario where the transformer_input table has the same row_id as the prediction_table?
In such case, what is the expectation?

Reply:
i'm mainly thinking about how the transformer inputs will be logged and how users will retrieve them at training time. will the joining logic need to be specified twice - once in the ST configuration for serving and again at training time? Does it make sense to include a field to identify which column in the prediction table corresponds to the row ID in the transformer input table?

Reply:
> will the joining logic need to be specified twice - once in the ST configuration for serving and again at training time?
Not quite sure if we need to specify the joining logic again in training time if it's already done in ST. In such case, I assume the prediction logs emitted by ST already has a complete table to be logged. 

> Does it make sense to include a field to identify which column in the prediction table corresponds to the row ID in the transformer input table?
Specific to prediction logging? We can, alternatively users can pre-join to create a table that needs to be logged.

Reply:
I know the final prediction table passed to the model will be logged, but we may also need to log some raw or intermediate features pre-transformation. are you saying the user will be able to pick and choose which tables to log from the ST? I think that could work. Will ST logging always be post-join (so all features are at the same granularity as the prediction table) or can also be pre-join?

Reply:
> are you saying the user will be able to pick and choose which tables to log from the ST? 

Yes, I think this should be supported and probably the main method of logging when standard transformer is used. 

> Will ST logging always be post-join (so all features are at the same granularity as the prediction table) or can also be pre-join? 

Post-join will be more preferable. With pre-join users still have to somehow give ST hint how to join it to form a complete table.

Reply:
ok, sounds good. anyway, can discuss this separately. this MR lgtm

Reply:
Thanks for the feedback @peterjrichens !